core/serialization/i_plugin_serializer.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The record type for an object containing functions for
+ *     serializing part of the workspace.
+ */
+
+'use strict';
+
+goog.module('Blockly.serialization.IPluginSerializer');
+goog.module.declareLegacyNamespace();
+
+// eslint-disable-next-line no-unused-vars
+const Workspace = goog.requireType('Blockly.Workspace');
+
+
+/**
+ * Serializes and deserializes a plugin.
+ * @record
+ */
+class IPluginSerializer {
+  constructor() {
+    /**
+     * A priority value used to determine the order of deserializing plugins.
+     * More positive priorities are deserialized before less positive
+     * priorities. Eg if you have priorities (0, -10, 10, 100) the order of
+     * deserialiation will be (100, 10, 0, -10).
+     * If two plugins have the same priority, they are deserialized in an
+     * arbitrary order relative to eachother.
+     * @type {number}
+     */
+    this.priority;
+  }
+
+  /* eslint-disable no-unused-vars, valid-jsdoc */
```

Comment:
`no-unused-vars` is because of the unused function parameters. `valid-jsdoc` is because `save` has an `@return` annotation, but no `return` statement.